### PR TITLE
Add hosts object for Animal Quest

### PR DESCRIPTION
**Pull request by jinitsuga**
adding a list of hosts (only Maya for now) to this package, as suggested by pjeweb on issue #246 from the website (found [here](https://github.com/alveusgg/alveusgg/pull/310))

### CODE REVIEW

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
You'll want to update the type of `host` in `AnimalQuest` to be a key of `hosts`, so that you get type-safety

**Comment by jinitsuga** [comment]
> You'll want to update the type of `host` in `AnimalQuest` to be a key of `hosts`, so that you get type-safety
This should take care of hosts type. Added `hostNames` object for whenever another host is added, so Typescript lets you know the host info should be added to `hosts` as well. Tried to keep it as robust as possible!

File: src/animal-quest.ts
Comment (MattIPv4) [inline src/animal-quest.ts]:
```suggestion
type Host = { name: string; link?: string };
type HostKey = keyof typeof hosts;

export const hosts = {
  maya: {
    name: "Maya Higa",
    link: "https://www.alveussanctuary.org/about/maya",
  },
} as const satisfies Record<string, Host>;
```
You've got a lot of duplicate data here, which we can simplify

File: src/animal-quest.ts
Comment (MattIPv4) [inline src/animal-quest.ts]:
```suggestion
  host: HostKey;
```
Instead of having the name here, now that we have a const/enum defining the hosts, we should reference it by key

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: src/animal-quest.ts
Comment (jinitsuga) [reply src/animal-quest.ts]: Ok! I implemented your suggestions and it's much shorter and clearer,  though with a slight change: made `HostKey` show actual hosts' name instead of just their key (would've have rendered just "maya" on the page otherwise), and renamed it to `HostName`.

Comment (MattIPv4) [reply src/animal-quest.ts]: We should use `HostKey`, and then update the page to do a lookup. Otherwise, you have the same problem you already have that there is no way to map the episode host back to further data about them...

Comment (jinitsuga) [reply src/animal-quest.ts]: Ahh I see! I sent the changes you suggested. I'll implement the website changes when it's up.
Thank you for the patience and apologies for the chaos :)

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
👀 It looks like you've introduced a rather unexpected `pnpm-lock.yaml` file, could you remove that?

You'll also want to run the command from the README to bump the package version to the next minor version

**Review by MattIPv4 — APPROVED** [review]
LGTM

**Comment by MattIPv4** [comment]
Would you mind creating a separate PR in the website repo that just bumps the data version and updates the animal quest pages to pull from this new object, just so any further data updates aren't blocked until your full hover card PR is merged?

Ty for the work on this!

**Comment by jinitsuga** [comment]
> 

For sure! I'll get at it tomorrow.
And thank YOU for the patience and explanations!

**Comment by jinitsuga** [comment]
> Would you mind creating a separate PR in the website repo that just bumps the data version and updates the animal quest pages to pull from this new object, just so any further data updates aren't blocked until your full hover card PR is merged?
> 
> Ty for the work on this!

Don't wanna cause more unnecessary chaos so I'll just ask: 
Since I'm apparently unable to fork `alveusgg` more than once(already forked and `main` branch is the hover cards PR) -- should I make a separate branch, reset to a commit prior to my work for the cards and work on these changes from there and PR those?
Thanks for any help :smile_cat:

**Comment by MattIPv4** [comment]
Yes, that would be the standard development flow, a branch for each feature you're working on.

**Pull request merged**
